src/background_remover.cc

```diff
@@ -268,8 +268,38 @@ static cv::Mat resizeAndPadTo(const cv::Mat &frame, int targetw, int targeth, Pa
     return ret;
 }
 
-void BackgroundRemover::maskBackground(cv::Mat &frame /* rgb */, const cv::Mat &maskImage /* rgb */,
-                                       bool do_blur_mask, Timing &t) {
+void BackgroundRemover::blendPixels(const cv::Vec3b &a,
+                                    const cv::Vec3b &b,
+                                    uint8_t p,
+                                    cv::Vec3b &result) {
+    for (int c = 0; c < 3; c++) {
```

Comment:
You shouldn’t need to iterate here. Looking at opencv/modules/core/include/opencv2/core/matx.hpp, operator* (for example) is implemented as:

```
template<typename _Tp, int cn> static inline
Vec<_Tp, cn>& operator *= (Vec<_Tp, cn>& a, int alpha)
{
    for( int i = 0; i < cn; i++ )
        a[i] = saturate_cast<_Tp>(a[i]*alpha);
    return a;
}
```

Reply:
As discussed on IRC, this seems to be the least messy way to do this :-D
I'll replace the hardcoded 3 with the .channels attribute of the vector...